SystemServices/SystemServices.cpp

```diff
@@ -1398,7 +1398,12 @@ namespace WPEFramework {
                     LOGWARN("request to switch to mode '%s' from mode '%s' \
                             with duration %d\n", newMode.c_str(),
                             oldMode.c_str(), duration);
-
+                     if(newMode == m_currentMode)
```

Comment:
Should we be checking for the same mode. We should check if existing setMode is in progress and the duration has not expired. So this should error out even if we set a different mode but before the previous setMode has expired.